ncs/encrypt_script.py

```diff
@@ -232,148 +181,82 @@ def generate_suit_encryption_info(self, iv, encrypted_cek, string_key_id, output
-
-
-def create_subparsers(parser):
-    """Create subparsers for the main parser.
-
-    This function adds subparsers for different commands to the main parser.
-    """
-    subparsers = parser.add_subparsers(dest="command", required=True, help="Choose subcommand:")
-
-    create_encrypt_and_generate_subparser(subparsers)
-    create_generate_subparser(subparsers)
-
-
-if __name__ == "__main__":
-    parser = ArgumentParser(
-        description="""This script allows to output artifacts needed by a SUIT envelope for encrypted firmware.
-
-It has two modes of operation:
-    - encrypt-and-generate: First encrypt the payload, then generate the files.
-    - generate: Only generate files based on encrypted firmware and the encrypted content/asset encryption key.
-    Note the encrypted firmware should match the format iv|tag|encrypted_firmware
-
-In both cases the output files are:
-    encrypted_content.bin - encrypted content of the firmware concatenated with the tag (encrypted firmware|16 byte tag).
-                            This file is used as the payload in the SUIT envelope.
-    suit_encryption_info.bin - The binary contents which should be included in the SUIT envelope as the contents of the suit-encryption-info parameter.
-
-Additionally, the encrypt-and-generate mode generates the following file:
-    plain_text_digest.bin - The digest of the plaintext firmware.
-    plain_text_size.txt - The size of the plaintext firmware in bytes.
-    """,  # noqa: W291, E501
-        formatter_class=RawTextHelpFormatter,
-    )
-
-    create_subparsers(parser)
-
-    arguments = parser.parse_args()
-
-    encrypted_asset = None
-    encrypted_cek = None
-
-    encryptor = Encryptor(arguments.kw_alg)
-
-    if arguments.command == "encrypt-and-generate":
-        encryptor.init_kms_backend(arguments.kms_script, arguments.context)
-        digest_generator = DigestGenerator(arguments.hash_alg.value)
-        digest_generator.generate_digest_size_for_plain_text(arguments.firmware, arguments.output_dir)
-        encrypted_asset, encrypted_cek = encryptor.generate_kms_artifacts(
-            arguments.firmware, arguments.key_name, arguments.context
+        encryption_info = self.generate_suit_encryption_info(init_vector, encrypted_cek, key_id)
+        return encrypted_content, tag, encryption_info
+
+    def _kw_alg_convert(self, kw_alg):
+        if kw_alg == SuitKWAlgorithms.A256KW:
+            self.cose_kw_alg = SuitEncryptAlgorithms.COSE_ALG_A256KW.value
+        else:
+            self.cose_kw_alg = SuitEncryptAlgorithms.COSE_ALG_DIRECT.value
+
+    def encrypt_and_generate(
+        self,
+        firmware: bytes,
+        key_name: str,
+        key_id: int,
+        context: str,
+        hash_alg: SuitDigestAlgorithms,
+        kw_alg: SuitKWAlgorithms,
+        kms_script: Path,
+    ) -> tuple[bytes, bytes, bytes, bytes, int]:
+        """
+        Encrypt the payload and generate the files.
```

Comment:
Same here.

Reply:
Fixed

Reply:
There were a couple of other places, also fixed them